File: YrradiaSDL2/Game/GUICore/GUILabel.cs
namespace YrradiaSDL2.Game.GUICore
{
    public class GUILabel : GUIComponent
    {
        private string m_text;
        private bool m_centerAligned;
        
        public GUILabel(string text, PointF position, bool centerAligned = false)
        {
            m_text = text;
            m_position = position;
            m_centerAligned = centerAligned;
        }

        protected override void RenderDerived()
        {
            base.RenderDerived();

            _.textRenderer.DrawString(m_text, m_position, Colors.wheat, FontSizes._20, m_centerAligned);
        }
    }
}

File: Desktop/code/client/JewelryMakerTheme/src/ForradiaEngine/GUICore/GUILabel.cpp
/*********************************************************************
 * Copyright 2026 Andreas Åkerberg                                   *
 * This code is licensed under MIT license (see LICENSE for details) *
 *********************************************************************/

#include "GUILabel.hpp"
#include "ForradiaEngine/Rendering/Text/TextRenderer.hpp"

namespace ForradiaEngine
{
    auto GUILabel::renderDerived() const -> void
    {
        auto bounds{dynamic_cast<const GUIComponent *>(this)->getBounds()};

        if (m_centerAlign)
        {
            bounds.x += bounds.width / 2;
            bounds.y += bounds.height / 2;
        }

        TextRenderer::instance().drawString(k_renderIDText, m_text, bounds.x, bounds.y,
                                            FontSizes::_20, m_centerAlign, true, m_color);
    }
}
